src/argilla/client/feedback/schemas.py

```diff
@@ -199,42 +205,21 @@ def set_suggestions(
             if not isinstance(suggestion, SuggestionSchema):
                 suggestion = SuggestionSchema(**suggestion)
             parsed_suggestions.append(suggestion)
-        if not self.id:
-            warnings.warn(
-                "Ignore the following if you are creating a new `FeedbackDataset` with"
-                " `FeedbackRecord`s, or if you are just working with a `FeedbackRecord`."
-                " Otherwise, if the `FeedbackRecord` is already pushed"
-                " to Argilla, note that `suggestions` have been provided, but the `id`"
-                " is not set, which means that the `FeedbackRecord` has been pushed to"
-                " Argilla, but hasn't been fetched, so the `id` is missing. To solve that,"
-                " you can simply call `FeedbackDataset.fetch_records()` to fetch them and"
-                " automatically set the `id`, to call `set_suggestions` on top of that."
-            )
+
+        suggestions_dict = {suggestion.question_name: suggestion for suggestion in self.suggestions}
         for suggestion in parsed_suggestions:
-            for existing_suggestion in self.suggestions:
-                if (
-                    existing_suggestion.question_id == suggestion.question_id
-                    or existing_suggestion.question_name == suggestion.question_name
-                ):
-                    warnings.warn(
-                        f"A suggestion for question `{existing_suggestion.question_name}` has already"
-                        " been provided, so the provided suggestion will overwrite it."
-                    )
-                    self.suggestions.remove(existing_suggestion)
-                    break
-        self.suggestions += parsed_suggestions
+            if suggestion.question_name in suggestions_dict:
+                warnings.warn(
+                    f"A suggestion for question `{suggestion.question_name}` has already"
+                    " been provided, so the provided suggestion will overwrite it.",
+                    category=UserWarning,
+                )
+            suggestions_dict[suggestion.question_name] = suggestion
```

Comment:
Even though this works, isn't it better to just keep the first one we find among the ones provided in the scenario mentioned at https://github.com/argilla-io/argilla/pull/3428, also including a warning so that the use knows what's happening? To differentiate between the scenario where the `suggestion` was already there, versus the scenario where the user provided two suggestions for the same question within `set_suggestions`? LMKWYT